x-pack/plugins/cases/public/components/markdown_editor/plugins/lens/plugin.tsx

```diff
@@ -0,0 +1,451 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { first } from 'rxjs/operators';
+import {
+  EuiFieldText,
+  EuiModalBody,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiMarkdownEditorUiPlugin,
+  EuiMarkdownContext,
+  EuiCodeBlock,
+  EuiSpacer,
+  EuiModalFooter,
+  EuiButtonEmpty,
+  EuiButton,
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiFormRow,
+  EuiMarkdownAstNodePosition,
+  EuiBetaBadge,
+} from '@elastic/eui';
+import React, { ReactNode, useCallback, useContext, useMemo, useEffect, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { useLocation } from 'react-router-dom';
+import styled from 'styled-components';
+
+import type { TypedLensByValueInput } from '../../../../../../lens/public';
+import { useKibana } from '../../../../common/lib/kibana';
+import { LensMarkDownRenderer } from './processor';
+import { DRAFT_COMMENT_STORAGE_ID, ID } from './constants';
+import { CommentEditorContext } from '../../context';
+import { ModalContainer } from './modal_container';
+import type { EmbeddablePackageState } from '../../../../../../../../src/plugins/embeddable/public';
+import {
+  SavedObjectFinderUi,
+  SavedObjectFinderUiProps,
+} from '../../../../../../../../src/plugins/saved_objects/public';
+import { useLensDraftComment } from './use_lens_draft_comment';
+
+const BetaBadgeWrapper = styled.span`
+  display: inline-flex;
+
+  .euiToolTipAnchor {
+    display: inline-flex;
+  }
+`;
+
+type LensIncomingEmbeddablePackage = Omit<EmbeddablePackageState, 'input'> & {
+  input: TypedLensByValueInput;
+};
+
+type LensEuiMarkdownEditorUiPlugin = EuiMarkdownEditorUiPlugin<{
+  title: string;
+  timeRange: TypedLensByValueInput['timeRange'];
+  startDate: string;
+  endDate: string;
+  position: EuiMarkdownAstNodePosition;
+  attributes: TypedLensByValueInput['attributes'];
+}>;
+
+interface LensSavedObjectsPickerProps {
+  children: ReactNode;
+  onChoose: SavedObjectFinderUiProps['onChoose'];
+}
+
+const LensSavedObjectsPickerComponent: React.FC<LensSavedObjectsPickerProps> = ({
+  children,
+  onChoose,
+}) => {
+  const { savedObjects, uiSettings } = useKibana().services;
+
+  const savedObjectMetaData = useMemo(
+    () => [
+      {
```

Comment:
this all looks static, no props? can move outside of component to a plain `const`